PathPlanning/DStarLite/d_star_lite.py

```diff
@@ -62,32 +65,46 @@ def __init__(self, ox: list, oy: list):
         self.y_max = int(abs(max(oy) - self.y_min_world))
         self.obstacles = [Node(x - self.x_min_world, y - self.y_min_world)
                           for x, y in zip(ox, oy)]
+        self.obstacles_xy = np.array(
+            [[obstacle.x, obstacle.y] for obstacle in self.obstacles]
+        )
         self.start = Node(0, 0)
         self.goal = Node(0, 0)
         self.U = list()  # type: ignore
         self.km = 0.0
         self.kold = 0.0
-        self.rhs = list()  # type: ignore
-        self.g = list()  # type: ignore
+        self.rhs = np.full(
+            (self.x_max, self.y_max),
+            float("inf")  # type: ignore
+        )
+        self.g = np.full(
+            (self.x_max, self.y_max),
+            float("inf")  # type: ignore
+        )
         self.detected_obstacles = list()  # type: ignore
+        self.detected_obstacles_xy = np.array([])
         if show_animation:
             self.detected_obstacles_for_plotting_x = list()  # type: ignore
             self.detected_obstacles_for_plotting_y = list()  # type: ignore
 
     def create_grid(self, val: float):
-        grid = list()
-        for _ in range(0, self.x_max):
-            grid_row = list()
-            for _ in range(0, self.y_max):
-                grid_row.append(val)
-            grid.append(grid_row)
-        return grid
+        return np.full((self.x_max, self.y_max), val)
 
     def is_obstacle(self, node: Node):
-        return any([compare_coordinates(node, obstacle)
-                    for obstacle in self.obstacles]) or \
-               any([compare_coordinates(node, obstacle)
-                    for obstacle in self.detected_obstacles])
+        x = np.array([node.x])
+        y = np.array([node.y])
+        is_in_obstacles = (
+            (self.obstacles_xy[:, 0] == x)
+            & (self.obstacles_xy[:, 1] == y)
+        ).any()
+        is_in_detected_obstacles = False
+        if self.detected_obstacles_xy.shape[0] > 0:
+            is_in_detected_obstacles = (
+                (self.detected_obstacles_xy[:, 0] == x)
+                & (self.detected_obstacles_xy[:, 1] == y)
```

Comment:
Difficult to read each condition, is it possible to improve it?

Reply:
I separated the conditions with boolean variables.